modules/core/shared/src/test/scala/retry/PackageObjectSpec.scala

```diff
@@ -216,6 +216,192 @@ class PackageObjectSpec extends AnyFlatSpec {
     assert(gaveUp)
   }
 
+  behavior of "retryingOnFailuresAndSomeErrors"
+
+  it should "retry until the action succeeds" in new TestContext {
+    val policy = RetryPolicies.constantDelay[StringOr](1.second)
+
+    val finalResult = retryingOnFailuresAndSomeErrors[String](
+      policy,
+      _.exists(_.toInt > 3),
```

Comment:
Not sure what you were intending here. This is the `wasSuccessful: String => Boolean` predicate. So on the third attempt it will be passed the string `"yay`". That string does pass this predicate, but it makes for a bit of a cryptic test 😄

Reply:
To be honest I was just trying to mimic some of the other tests in the suite. I can update these to a less cryptic form.